var/spack/repos/builtin/packages/python/package.py

```diff
@@ -168,12 +173,33 @@ def configure_args(self):
 
         # setup.py needs to be able to read the CPPFLAGS and LDFLAGS
         # as it scans for the library and headers to build
-        dep_pfxs = [dspec.prefix for dspec in spec.dependencies('link')]
-        config_args = [
-            '--with-threads',
-            'CPPFLAGS=-I{0}'.format(' -I'.join(dp.include for dp in dep_pfxs)),
-            'LDFLAGS=-L{0}'.format(' -L'.join(dp.lib for dp in dep_pfxs)),
-        ]
+        link_deps = spec.dependencies('link')
+
+        # Header files are often included assuming they reside in a
+        # subdirectory of prefix.include, e.g. #include <openssl/ssl.h>,
+        # which is why we don't use HeaderList here. The header files of
+        # libffi reside in prefix.lib but the configure script of Python
+        # finds them using pkg-config.
+        cppflags = '-I' + ' -I'.join(dep.prefix.include
+                                     for dep in link_deps
+                                     if dep.name != 'libffi')
+
+        # Currently, the only way to get SpecBuildInterface wrappers of the
+        # dependencies (which we need to get their 'libs') is to get them
+        # using spec.__getitem__.
+        ldflags = ' '.join(spec[dep.name].libs.search_flags
+                           for dep in link_deps)
+
+        config_args = ['CPPFLAGS=' + cppflags, 'LDFLAGS=' + ldflags]
+
+        # https://docs.python.org/3/whatsnew/3.7.html#build-changes
+        if spec.satisfies('@:3.6'):
+            config_args.append('--with-threads')
+
+        if '^libffi' in spec:
+            config_args.append('--with-system-ffi')
+        else:
```

Comment:
how would this case ever be triggered?

Reply:
When `libffi` is among the dependencies, which is the case `when="@3.7:"`.

Reply:
obviously... sorry